handler/middleware.go

```diff
@@ -181,33 +191,21 @@ func (m *middleware) Close(ctx context.Context) error {
 }
 
 type guest struct {
-	ns               wazero.Namespace
 	guest            wazeroapi.Module
 	handleRequestFn  wazeroapi.Function
 	handleResponseFn wazeroapi.Function
 }
 
 func (m *middleware) newGuest(ctx context.Context) (*guest, error) {
-	ns, err := m.newNamespace(ctx, m.runtime)
-	if err != nil {
-		return nil, fmt.Errorf("wasm: error creating namespace: %w", err)
-	}
-
-	// Note: host modules don't use configuration
-	_, err = ns.InstantiateModule(ctx, m.hostModule, wazero.NewModuleConfig())
-	if err != nil {
-		_ = ns.Close(ctx)
-		return nil, fmt.Errorf("wasm: error instantiating host: %w", err)
-	}
-
-	g, err := ns.InstantiateModule(ctx, m.guestModule, m.moduleConfig)
+	g, err := m.runtime.InstantiateModule(ctx, m.guestModule, m.moduleConfig.
+		// TODO: use true random name / or make change to wazero to allow anonymous module.
+		WithName(fmt.Sprintf("%#x", time.Now().UnixNano())))
```

Comment:
@codefromthecrypt @anuraaga this makes me believe that we should add support for this - instantiating without names. I think this is the common use case and actually, some users asked about this "already instantiated" error before. wdyt?

Reply:
usually we suggest using an atomic https://github.com/wapc/wapc-go/blob/master/engines/wazero/wazero.go#L359